girder-dandi-archive/girder_dandi_archive/__init__.py

```diff
@@ -19,6 +20,9 @@ def load(self, info):
             {"$setOnInsert": {"value": 1}},
             upsert=True,
         )
+        # Index for rest.DandiResource.get_dandiset and other endpoints that might
+        # require an index on meta.dandiset.identifier
```

Comment:
```suggestion
        # Index used by rest.DandiResource.get_dandiset and others.
        # This compound index is necessary, otherwise MongoDB seems to use only the
        # parentId index instead of a plain meta.dandiset.identifier index, given a
        # query for both fields.
```

I think it's most important that we communicate the *why* instead of the *what* in these sort of code comments. Any competent developer can figure out that this is an index and it's used for queries on the `meta.dandiset.identifier` field just by inspection of the code itself. Harder questions to answer from the code itself are:
1. *"Is this fragment still used? Where?"* - This could be answered by a global search, but given that the code for a dandiset query is split among two separate places (this index initialization and the query itself), this is a helpful (but not essential) comment.
2. *"Why is this a compound index? There's already an index on `parentId`, so I'll just simplify this index again..."* - This is definitely not obvious from looking at the code itself. We need to preserve the in-depth analysis that went into this programming decision, so others don't break it in the future.

Reply:
I rewrote my comment and tried to answer the questions you posed.